templates/opencodelists/user_create_codelist.html

```diff
@@ -3,12 +3,89 @@
 {% block title_extra %}Create a new codelist{% endblock %}
 
 {% block content %}
-  {% #registration_form button_text="Create" csrf_token=csrf_token form=form title="Create a new codelist" enctype="multipart/form-data"%}
-    {% if form.owner %}
-      {% form_input field=form.owner %}
+  <form
+    class="flex gap-6 min-h-full flex-col justify-center py-20 sm:mx-auto sm:w-full sm:max-w-3xl sm:px-6 lg:px-8"
+    enctype="multipart/form-data"
+    method="POST"
+  >
+    <div class="sm:mx-auto text-center">
+      <h1 class="text-2xl/9 font-bold tracking-tight text-slate-900">
+        Create a new codelist
+      </h1>
+    </div>
+
+    {% csrf_token %}
+
+    {% if form.non_field_errors %}
+      <div class="rounded-md bg-red-50 p-4">
+        <div class="flex">
+          <div class="shrink-0">
+            {% icon_exclamation_circle_mini class="size-5 text-red-400" %}
+          </div>
+          <div class="ml-3">
+            <h3 class="text-sm font-medium text-red-800">{{ error_title }}</h3>
+            <div class="text-sm text-red-700">
+              <ul role="list" class="list-disc space-y-1 pl-5">
+                {% for error in form.non_field_errors %}
+                  <li>{{ error }}</li>
+                {% endfor %}
+              </ul>
+            </div>
+          </div>
+        </div>
+      </div>
     {% endif %}
-    {% form_input field=form.name %}
-    {% form_input field=form.coding_system_id %}
-    {% form_input_upload field=form.csv_data %}
-  {% /registration_form %}
+
+    <div class="bg-white px-6 py-12 flex flex-col gap-6 shadow-sm sm:rounded-lg sm:px-12">
+      {% if form.owner %}
+        {% form_input field=form.owner %}
+      {% endif %}
+
+      {% fragment as name_help_text %}
+        <p>Describe the purpose of this codelist in 255 characters or fewer.</p>
+        <p>You should use sentence case and spaces between words, rather than underscores or hyphens. Being descriptive is more important than being concise.</p>
+        <p>For example: <strong>"Type 2 diabetes without retinopathy"</strong> is a better name than <strong>"diabetes"</strong>.</p>
+      {% endfragment name_help_text %}
+      {% form_input field=form.name help_text=name_help_text %}
+    </div>
+
+    <div class="bg-white px-6 py-12 shadow-sm sm:rounded-lg sm:px-12">
+      {% fragment as coding_system_help_text %}
+        <ul>
+          <li><strong>CTV3 (Read v3)</strong>: Clinical events, drugs, and appliances (historic; Primary and Secondary Care)</li>
```

Comment:
Can't help feeling that these descriptions would be better implemented as properties of their respective coding system classes (as part of the `BuilderCompatibleCodingSystem` interface) rather than hard coding them in the template.